src/qt/utilitydialog.cpp

```diff
@@ -144,17 +142,16 @@ ShutdownWindow::ShutdownWindow(QWidget *parent, Qt::WindowFlags f):
     setLayout(layout);
 }
 
-QWidget *ShutdownWindow::showShutdownWindow(BitcoinGUI *window)
+QWidget* ShutdownWindow::showShutdownWindow(QWidget* parent)
 {
-    if (!window)
-        return nullptr;
+    assert(parent != nullptr);
 
     // Show a simple window indicating shutdown status
     QWidget *shutdownWindow = new ShutdownWindow();
-    shutdownWindow->setWindowTitle(window->windowTitle());
+    shutdownWindow->setWindowTitle(parent->windowTitle());
```

Comment:
nit: there is no guarantee now, that `parent` is a top-level widget. 

May I suggest a following diff:
```diff
diff --git a/src/qt/utilitydialog.cpp b/src/qt/utilitydialog.cpp
index 65e8823cb..a1ce8e741 100644
--- a/src/qt/utilitydialog.cpp
+++ b/src/qt/utilitydialog.cpp
@@ -19,10 +19,12 @@
 
 #include <QCloseEvent>
 #include <QLabel>
+#include <QMainWindow>
 #include <QRegExp>
 #include <QTextTable>
 #include <QTextCursor>
 #include <QVBoxLayout>
+#include <QWidget>
 
 /** "Help message" or "About" dialog box */
 HelpMessageDialog::HelpMessageDialog(interfaces::Node& node, QWidget *parent, bool about) :
@@ -142,16 +144,16 @@ ShutdownWindow::ShutdownWindow(QWidget *parent, Qt::WindowFlags f):
     setLayout(layout);
 }
 
-QWidget* ShutdownWindow::showShutdownWindow(QWidget* parent)
+QWidget* ShutdownWindow::showShutdownWindow(QMainWindow* window)
 {
-    assert(parent != nullptr);
+    assert(window);
 
     // Show a simple window indicating shutdown status
     QWidget *shutdownWindow = new ShutdownWindow();
-    shutdownWindow->setWindowTitle(parent->windowTitle());
+    shutdownWindow->setWindowTitle(window->windowTitle());
 
     // Center shutdown window at where main window was
-    const QPoint global = parent->mapToGlobal(parent->rect().center());
+    const QPoint global = window->mapToGlobal(window->rect().center());
     shutdownWindow->move(global.x() - shutdownWindow->width() / 2, global.y() - shutdownWindow->height() / 2);
     shutdownWindow->show();
     return shutdownWindow;
diff --git a/src/qt/utilitydialog.h b/src/qt/utilitydialog.h
index b52c26758..fdf1bd385 100644
--- a/src/qt/utilitydialog.h
+++ b/src/qt/utilitydialog.h
@@ -6,6 +6,7 @@
 #define BITCOIN_QT_UTILITYDIALOG_H
 
 #include <QDialog>
+#include <QMainWindow>
 #include <QWidget>
 
 namespace interfaces {
@@ -44,7 +45,7 @@ class ShutdownWindow : public QWidget
 
 public:
     explicit ShutdownWindow(QWidget *parent=nullptr, Qt::WindowFlags f=Qt::Widget);
-    static QWidget* showShutdownWindow(QWidget* window);
+    static QWidget* showShutdownWindow(QMainWindow* window);
 
 protected:
     void closeEvent(QCloseEvent *event
```

Reply:
@hebasto considering https://doc.qt.io/qt-5/qwidget.html#top-level-and-child-widgets how about change the assertion to
```cpp
    assert(parent != nullptr && parent->parentWidget() == nullptr);
```

Reply:
Yes, this will work.
But using `QMainWindow` as a parameter type seems clearer semantically (as `BitcoinGUI` is a direct sub-class of `QMainWindow`), and makes checking `parent->parentWidget() == nullptr` redundant, no?

Reply:
Now using `QMainWindow`, I think it's unnecessary to depend on `QMainWindow` just to get the title and rect but at the same time it's no big deal.